devices/r4isdhc.cpp

```diff
@@ -171,32 +171,92 @@ bool writeNor(const uint32_t dest_address, const uint32_t length, const uint8_t
     return true;
 }
 
-bool trySecureInit(BlowfishKey key) {
-    union {
-        uint32_t u32;
-        uint8_t bytes[4];
-    } buf;
+bool sendRawInit() {
+    if (ntrcard::state.status != ntrcard::Status::RAW) {
+        logMessage(LOG_ERR, "r4isdhc: sendRawInit: status (%d) not RAW",
+            static_cast<uint32_t>(ntrcard::state.status));
+        return false;
+    }
+
+    CmdBuf4 buf;
+    // this is actually the NOR write disable command
+    // the r4isdhc will respond to cart commands with 0xFFFFFFFF if
+    // the "magic" command hasn't been sent, so we check for that
+    sendCommand(0x40199, 4, buf.u8, 0x180000);
+    if (buf.u32 != 0xFFFFFFFF) {
+        logMessage(LOG_ERR, "r4isdhc: sendRawInit: pre-test returned 0x%08X", buf.u32);
+        return false;
+    }
+
+    if (platform::CAN_RESET) {
+        // actually if we're here CAN_RESET is true
+        // but.. can't be too safe
+        ntrcard::init();
+    }
+    sendCommand(0x68, 4, nullptr, 0x180000);
+    // now it will return zeroes
+    sendCommand(0x40199, 4, buf.u8, 0x180000);
+    if (buf.u32 == 0) {
+        return true;
+    }
+
+    logMessage(LOG_ERR, "r4isdhc: sendRawInit: post-test returned 0x%08X", buf.u32);
+    return false;
+}
+
+bool sendKey2Init() {
```

Comment:
this function has a little-confused name.
how about change like `unlockNorWriteOld` (or Type2. samely sendRawInit also can change..)

Reply:
99 01 04 is Write Disable. It's only used as a test command, it isn't actually needed in initialisation.

Write Enable is sent elsewhere.

Reply:
okey, so how about rename `testTypeXXX`?

Reply:
But it's not a test, it's actually initialising the cart and then verifying that it succeeded.

Reply:
okay `verify` whatever, it's more clear reading. we need clearly result meaning not the real calling action chain.

if that action will do initialize then verify, it just test or verify something.
but just add `send` prefix, it can be confused.

Reply:
I don't think that anyone, when they see `sendKey2Init` _in this context_ that returns a boolean-ish value, will be confused as to what the function is doing.

But OK, sure.